addons/mail/models/mail_thread.py

```diff
@@ -2881,8 +2890,19 @@ def _message_create(self, values_list):
             for x in ('from', 'to', 'cc'):
                 create_values.pop(x, None)
             create_values['partner_ids'] = [Command.link(pid) for pid in (create_values.get('partner_ids') or [])]
+            channel_ids = create_values.pop('channel_ids', None)
             create_values_list.append(create_values)
 
+            # Create Notification in mentioned channels
+            if channel_ids and (self._name != 'discuss.channel' or self.channel_type == 'channel'):
+                body = Markup("%(partner)s %(text)s %(record)s") % {
+                    "partner": self.env.user.partner_id._get_html_link(),
+                    "text": _("mentioned this channel in"),
+                    "record": self._get_html_link(),
+                }
```

Comment:
This is not good for translation to split sentence like this. Name + "mention this channel in" + link is constraining sentence shape too much for other languages.

Also translators need clue in exported sentence with placeholders. If they see `$(partner)s %(text)s %(record)s` they won't understand what's the fuss with it. And no, they essentially don't look at the code because this is very inefficient.

So have exported sentence look like this: `%(partner)s mentioned this channel in %(record)s`.